lib/services/front.ts

```diff
@@ -37,25 +37,31 @@ export class FrontService extends ServiceScaffold<string> implements ServiceList
 
 	constructor(data: FrontConstructor | FrontListenerConstructor) {
 		super(data);
-		this.session = new Front(data.token);
 		if (data.type === ServiceType.Listener) {
 			const listenerData = <FrontListenerConstructor>data;
-			// This swallows webhook events.  When operating on an entire inbox we use its webhook rule,
-			// but a webhook still requires somewhere to send its webhooks to.
+			this.session = new Front(listenerData.token, listenerData.secret);
+			// This swallows webhook events associated with a channel.
+			// When operating on an entire inbox we use its webhook rule,
+			// but a channel still requires somewhere to send its webhooks to.
 			this.registerHandler('front-dev-null', (_formData, response) => {
 				response.sendStatus(200);
 			});
-			// Create an endpoint for this listener and enqueue events
-			this.registerHandler(listenerData.path || FrontService._serviceName, (formData, response) => {
-				this.queueData({
-					context: formData.body.conversation.id,
-					cookedEvent: {},
-					rawEvent: formData.body,
-					source: FrontService._serviceName,
-					type: formData.body.type,
-				});
-				response.sendStatus(200);
+			this.session.registerEvents({
+				hookPath: `/${listenerData.path || FrontService._serviceName}`,
+				server: this.expressApp,
+			}, (error: Error | null, event?: Event) => {
```

Comment:
I know this is pre-existing, but why does this use a callback instead of a promise? Looking at the internal code it looks like bothe `registerEvents` and `httpCall` both use promises internally but use callbacks in their call signature.

Reply:
This uses a callback because it is registering a handler that may occur many times in response to external events, rather than a Promise which should only be completed once.